src/components/graphs/GraphContainer.js

```diff
@@ -18,14 +18,26 @@ import Pagination from './pagination/Pagination';
 // Assets
 import { stateData } from './assets/bargraphAssets';
 
-const filterData = (data, today, elevenMonths, state) => {
+const filterData = (
+  data,
+  today,
+  elevenMonths,
+  state,
+  dateIsMilli,
+  setDateIsMilli
+) => {
   // Only grab events that have occurred in the last 12 months and filter out any that do not have dates:
   const start = today - elevenMonths;
+  let dateFilter;
 
-  const dateFilter = data.map(incident => {
-    incident.date = Date.parse(incident.date);
-    return incident;
-  });
+  !dateIsMilli
```

Comment:
Can you use this before declaring it? I see it in the useState hook below, but I thought the hook would have to come first.

Reply:
I guess I'm wondering if all your state should be at the top of the page?

Reply:
What is being used before declaration?

Reply:
!dateIsMilli unless I'm reading it incorrectly

Reply:
Nevermind. It is part of the filter const.

Reply:
But should the state live at the top of the page?

Reply:
That is part of a helper function that lives outside of the component. At run time it exists inside the correct scope

Reply:
Gotcha Jason. Thanks for explaining.